torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -1247,6 +1261,13 @@ void ProcessGroupNCCL::workCleanupLoop() {
       auto& work = *it;
       work.checkAndSetException();
       bool timedOut = work.checkTimeout();
+      c10::optional<std::thread> dumpingDebugInfo;
+
+      if (shouldAutoDumpWorkTimeOut(work)) {
+        checkAndSetStore();
+        // Store debug info to storage. (By default to local disk)
+        dumpingDebugInfo = tryWriteDebugInfo();
```

Comment:
i think i am gonna land my PR first (well, i already told the bot but we'll see).  So you may want to rebase on mine to get used to the new future thing

Reply:
Yes, I definitely want to do rebase once.